cmd/cli/kubectl-kyverno/commands/fix/test/options.go

```diff
@@ -142,6 +143,53 @@ func (o options) execute(out io.Writer, dirs ...string) error {
 				fmt.Fprintln(out, "    OK")
 			}
 		}
+		if testCase.Test.Variables != "" {
+			fmt.Fprintf(out, "  Processing values file (%s)...\n", testCase.Test.Variables)
+			path := filepath.Join(testCase.Dir(), testCase.Test.Variables)
+			values, err := values.Load(nil, path)
+			if err != nil {
+				fmt.Fprintf(out, "    ERROR: failed to load values: %s\n", err)
+				continue
+			}
+			fixed, messages, err := fix.FixValues(*values)
+			for _, warning := range messages {
+				fmt.Fprintln(out, "  WARNING:", warning)
+			}
+			if err != nil {
+				fmt.Fprintln(out, "  ERROR:", err)
+				continue
+			}
+			needsSave := !reflect.DeepEqual(values, &fixed)
+			if o.save && (o.force || needsSave) {
+				fmt.Fprintf(out, "  Saving values file (%s)...\n", path)
+				untyped, err := kubeutils.ObjToUnstructured(fixed)
+				if err != nil {
+					fmt.Fprintf(out, "    ERROR: converting to unstructured: %s\n", err)
+					continue
+				}
+				unstructured.RemoveNestedField(untyped.UnstructuredContent(), "metadata", "creationTimestamp")
+				unstructured.RemoveNestedField(untyped.UnstructuredContent(), "metadata", "generation")
+				unstructured.RemoveNestedField(untyped.UnstructuredContent(), "metadata", "uid")
```

Comment:
How do we come to this conclusion?

Reply:
Those fields don't have `omitempty` we need to remove them manually.
Not removing them is not technically an error but it pollutes the file 🤷

Reply:
It tried to create schemas with `TypeMeta` only but it's not allowed, we need `ObjectMeta` too.
And `ObjectMeta` is a very special object in k8s, it's schema is managed internally by the apiserver, this is kind of dynamic.